content/news/XXXX-XX-XX-dvs-internals-pt3.md

```diff
@@ -0,0 +1,219 @@
+can be so large that the signal floats entirely above the x-axis for multiple
+cycles - making zero-crossing detection impossible.
+
+Even when that doesn’t happen, the offset causes the time interval $\Delta t$
+between zero-crossings to become irregular, introducing audible pitch flutter.
+
+To decode the signal, we must solve:
+
+1. How to filter the signal to enable pitch detection
+2. How to demodulate this non-standard modulation
+3. How to decipher the code that is modulated onto the carrier
+
+---
+
+## Pitch Detection
+
+> **Note:** If you're unfamiliar with pitch detection in DVS systems, revisit
+> [DVS Internals Pt. 1]({filename}/news/2021-11-21-dvs-internals-pt1.md).
+
+Our goal is to produce a signal that oscillates evenly around the x-axis. This
+filtered waveform can then be processed by the existing pitch detection algorithm.
+
+A simple discrete derivative operation achieves this:
+
+$$
+y[n] = x[n] - x[n-1] \tag{1}
+$$
+
+$\text{where:}$<br>
+$\text{- x[n]: Input sample}$<br>
+$\text{- x[n-1]: Delayed input sample}$<br>
+$\text{- y[n]: Difference of both values}$<br>
+<br>
+
+When applied to the offset-modulated signal, we get:
+
+![Traktor MK2 signal with derivative]({static}/images/news/traktor-mk2-signal-with-derivative.svg)
+
+The resulting waveform oscillates cleanly around zero, which is ideal for
+analysis. It also makes it easier to pinpoint the half-cycle peaks needed for
+bit detection.
+
+---
+
+## Demodulation Techniques
+
+To extract bits from the signal, we detect the zero-crossings and sample the
+amplitude of the sinusoid at those moments.
+
+![Traktor MK2 signal with
+zero-crossings]({static}/images/news/traktor-mk2-signal-with-zero-crossings.svg)
+
+You may notice that the derivative’s zero-crossings don’t align perfectly with
+the original peaks. That’s due to a delay introduced by the filter. Smoothing
+the signal first, then compensating for the delay (e.g., by selecting
+$x[n-3]$), yields better results.
+
+For greater accuracy, one could analyze the phase response $\phi(\omega)$,
+which shows how filter delay varies with input frequency - but for this use case,
+a fixed delay works well enough.
+
+The filtered signal can cross the x-axis in two directions-positive to
+negative or vice versa. Based on the direction, we determine which half-cycle
+contains the encoded bit.
+
+![Traktor MK2 signal with readings]({static}/images/news/traktor-mk2-signal-with-readings.svg)
+
+Demodulation is then as simple as applying a threshold: amplitudes above it
+are **1**, and below it are **0**.
+
+![Traktor MK2 signal with timecodes]({static}/images/news/traktor-mk2-signal-with-timecodes.svg)
+
+On actual vinyl, the physical behavior of the needle causes the offset to
+decay over time, because the needle slowly drifts back to the middle. This decay complicates bit extraction.
+
+To compensate, we analyze the **slope** between subsequent readings by
+reusing the derivative equation in $\text{(1)}$.
```

Comment:
by then encoding negative slope as 0 and positive slope as 1?

Reply:
Added a clearer description